### ISSUE TITLE
Cannot use hs.load()

### ISSUE THREAD

**Issue by lnaglecocco** [hyperspy/hyperspy#3212]
#### Describe the bug
I have been using Hyperspy for a few years now to analyse EELS data. However, now when I try to use it, the hs.load() method does not work. No error is reported, but it seems to go into an infinite loop, with my Jupyter notebook showing that the code is running indefinitely. I have tried this with data which I analysed successfully using Hyperspy in the past and I encounter this issue.

My guess is that either there is a bug in the latest version of Hyperspy (which seems unlikely) or there is some incompatibility in the packages I have locally.

#### To Reproduce

I have these import statements:

```
%matplotlib qt
import matplotlib
import matplotlib.pyplot as plt
import numpy as np
import hyperspy.api as hs
import pickle 
%config Completer.use_jedi = False
```

Then I load a file like so:
```
LowLoss_raw = hs.load("EELS Spectrum Image (low-loss).dm4")
```

And this hs.load statement gets stuck in an infinite loop

#### Expected behavior
I expect the code to run in <3 seconds and the data to be loaded.

#### Python environement:
Here are a full list of packages in my conda environment
```
aom                       3.5.0                h63175ca_0    conda-forge
asciitree                 0.3.3                      py_2    conda-forge
ase                       3.22.1             pyhd8ed1ab_1    conda-forge
asttokens                 2.2.1              pyhd8ed1ab_0    conda-forge
backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
backports                 1.0                pyhd8ed1ab_3    conda-forge
backports.functools_lru_cache 1.6.5              pyhd8ed1ab_0    conda-forge
blinker                   1.6.2              pyhd8ed1ab_0    conda-forge
blosc                     1.21.4               hdccc3a2_0    conda-forge
brotli                    1.0.9                hcfcfb64_9    conda-forge
brotli-bin                1.0.9                hcfcfb64_9    conda-forge
brotli-python             1.0.9           py310h00ffb61_9    conda-forge
bzip2                     1.0.8                h8ffe710_4    conda-forge
c-blosc2                  2.10.0               h183a6f4_0    conda-forge
ca-certificates           2023.7.22            h56e8100_0    conda-forge
cached-property           1.5.2                hd8ed1ab_1    conda-forge
cached_property           1.5.2              pyha770c72_1    conda-forge
certifi                   2023.7.22          pyhd8ed1ab_0    conda-forge
charls                    2.4.2                h1537add_0    conda-forge
charset-normalizer        3.2.0              pyhd8ed1ab_0    conda-forge
click                     8.1.7           win_pyh7428d3b_0    conda-forge
cloudpickle               2.2.1              pyhd8ed1ab_0    conda-forge
colorama                  0.4.6              pyhd8ed1ab_0    conda-forge
comm                      0.1.4              pyhd8ed1ab_0    conda-forge
contourpy                 1.1.0           py310h232114e_0    conda-forge
cycler                    0.11.0             pyhd8ed1ab_0    conda-forge
cytoolz                   0.12.2          py310h8d17308_0    conda-forge
dask-core                 2023.8.0           pyhd8ed1ab_0    conda-forge
dav1d                     1.2.1                hcfcfb64_0    conda-forge
debugpy                   1.6.8           py310h00ffb61_0    conda-forge
decorator                 5.1.1              pyhd8ed1ab_0    conda-forge
dill                      0.3.7              pyhd8ed1ab_0    conda-forge
distributed               2023.8.0           pyhd8ed1ab_0    conda-forge
entrypoints               0.4                pyhd8ed1ab_0    conda-forge
executing                 1.2.0              pyhd8ed1ab_0    conda-forge
fasteners                 0.17.3             pyhd8ed1ab_0    conda-forge
flask                     2.3.2              pyhd8ed1ab_0    conda-forge
fonttools                 4.42.0          py310h8d17308_0    conda-forge
freetype                  2.12.1               h546665d_1    conda-forge
fsspec                    2023.6.0           pyh1a96a4e_0    conda-forge
gettext                   0.21.1               h5728263_0    conda-forge
giflib                    5.2.1                h64bf75a_3    conda-forge
glib                      2.76.4               h12be248_0    conda-forge
glib-tools                2.76.4               h12be248_0    conda-forge
gst-plugins-base          1.22.5               h001b923_0    conda-forge
gstreamer                 1.22.5               hb4038d2_0    conda-forge
h5py                      3.9.0           nompi_py310h727aef9_100    conda-forge
hdf5                      1.14.0          nompi_h918d9b7_103    conda-forge
hyperspy                  1.7.5           py310h5588dad_1    conda-forge
hyperspy-base             1.7.5           py310h8d17308_1    conda-forge
hyperspy-gui-ipywidgets   1.5.0              pyhd8ed1ab_0    conda-forge
hyperspy-gui-traitsui     1.5.2              pyhd8ed1ab_0    conda-forge
icu                       72.1                 h63175ca_0    conda-forge
idna                      3.4                pyhd8ed1ab_0    conda-forge
imagecodecs               2023.8.12       py310h997794f_0    conda-forge
imageio                   2.22.0             pyhfa7a67d_0    conda-forge
importlib-metadata        6.8.0              pyha770c72_0    conda-forge
importlib_metadata        6.8.0                hd8ed1ab_0    conda-forge
importlib_resources       6.0.1              pyhd8ed1ab_0    conda-forge
intel-openmp              2023.2.0         h57928b3_49496    conda-forge
ipyfilechooser            0.6.0              pyhd8ed1ab_0    conda-forge
ipykernel                 6.23.1             pyh025b116_0    conda-forge
ipyparallel               8.6.1              pyhd8ed1ab_0    conda-forge
ipython                   8.14.0             pyh08f2357_0    conda-forge
ipywidgets                8.1.0              pyhd8ed1ab_0    conda-forge
itsdangerous              2.1.2              pyhd8ed1ab_0    conda-forge
jedi                      0.19.0             pyhd8ed1ab_0    conda-forge
jinja2                    3.1.2              pyhd8ed1ab_1    conda-forge
joblib                    1.3.2              pyhd8ed1ab_0    conda-forge
jupyter_client            8.3.0              pyhd8ed1ab_0    conda-forge
jupyter_core              5.3.1           py310h5588dad_0    conda-forge
jupyterlab_widgets        3.0.8              pyhd8ed1ab_0    conda-forge
jxrlib                    1.1                  h8ffe710_2    conda-forge
kiwisolver                1.4.4           py310h232114e_1    conda-forge
krb5                      1.20.1               heb0366b_0    conda-forge
lazy_loader               0.3                pyhd8ed1ab_0    conda-forge
lcms2                     2.15                 h3e3b177_1    conda-forge
lerc                      4.0.0                h63175ca_0    conda-forge
libaec                    1.0.6                h63175ca_1    conda-forge
libavif                   0.11.1               ha7b97ba_2    conda-forge
libblas                   3.9.0              17_win64_mkl    conda-forge
libbrotlicommon           1.0.9                hcfcfb64_9    conda-forge
libbrotlidec              1.0.9                hcfcfb64_9    conda-forge
libbrotlienc              1.0.9                hcfcfb64_9    conda-forge
libcblas                  3.9.0              17_win64_mkl    conda-forge
libclang                  16.0.6          default_heb8d277_1    conda-forge
libclang13                16.0.6          default_hc80b9e7_1    conda-forge
libcurl                   8.1.1                h68f0423_0    conda-forge
libdeflate                1.18                 hcfcfb64_0    conda-forge
libffi                    3.4.2                h8ffe710_5    conda-forge
libglib                   2.76.4               he8f3873_0    conda-forge
libhwloc                  2.9.2           nocuda_h15da153_1008    conda-forge
libiconv                  1.17                 h8ffe710_0    conda-forge
libjpeg-turbo             2.1.5.1              hcfcfb64_0    conda-forge
liblapack                 3.9.0              17_win64_mkl    conda-forge
libogg                    1.3.4                h8ffe710_1    conda-forge
libpng                    1.6.39               h19919ed_0    conda-forge
libsodium                 1.0.18               h8d14728_1    conda-forge
libsqlite                 3.42.0               hcfcfb64_0    conda-forge
libssh2                   1.11.0               h7dfc565_0    conda-forge
libtiff                   4.5.1                h6c8260b_0    conda-forge
libvorbis                 1.3.7                h0e60522_0    conda-forge
libwebp-base              1.3.1                hcfcfb64_0    conda-forge
libxcb                    1.15                 hcd874cb_0    conda-forge
libxml2                   2.11.5               hc3477c8_0    conda-forge
libzlib                   1.2.13               hcfcfb64_5    conda-forge
libzopfli                 1.0.3                h0e60522_0    conda-forge
link-traits               1.0.3              pyhd8ed1ab_3    conda-forge
llvmlite                  0.40.1          py310hb84602e_0    conda-forge
locket                    1.0.0              pyhd8ed1ab_0    conda-forge
lz4-c                     1.9.4                hcfcfb64_0    conda-forge
m2w64-gcc-libgfortran     5.3.0                         6    conda-forge
m2w64-gcc-libs            5.3.0                         7    conda-forge
m2w64-gcc-libs-core       5.3.0                         7    conda-forge
m2w64-gmp                 6.1.0                         2    conda-forge
m2w64-libwinpthread-git   5.0.0.4634.697f757               2    conda-forge
markupsafe                2.1.3           py310h8d17308_0    conda-forge
matplotlib                3.7.2           py310h5588dad_0    conda-forge
matplotlib-base           3.7.2           py310h51140c5_0    conda-forge
matplotlib-inline         0.1.6              pyhd8ed1ab_0    conda-forge
mkl                       2022.1.0           h6a75c08_874    conda-forge
mpmath                    1.3.0              pyhd8ed1ab_0    conda-forge
mrcz                      0.5.6              pyh9f0ad1d_1    conda-forge
msgpack-python            1.0.5           py310h232114e_0    conda-forge
msys2-conda-epoch         20160418                      1    conda-forge
munkres                   1.1.4              pyh9f0ad1d_0    conda-forge
natsort                   8.4.0              pyhd8ed1ab_0    conda-forge
nest-asyncio              1.5.6              pyhd8ed1ab_0    conda-forge
networkx                  3.1                pyhd8ed1ab_0    conda-forge
numba                     0.57.1          py310h19bcfe9_0    conda-forge
numcodecs                 0.11.0          py310h00ffb61_1    conda-forge
numexpr                   2.8.4           mkl_py310h98e78b8_1    conda-forge
numpy                     1.24.3          py310hd02465a_0    conda-forge
openjpeg                  2.5.0                ha2aaf27_2    conda-forge
openssl                   3.1.2                hcfcfb64_0    conda-forge
packaging                 23.1               pyhd8ed1ab_0    conda-forge
parso                     0.8.3              pyhd8ed1ab_0    conda-forge
partd                     1.4.0              pyhd8ed1ab_0    conda-forge
pcre2                     10.40                h17e33f8_0    conda-forge
pickleshare               0.7.5                   py_1003    conda-forge
pillow                    10.0.0          py310hb653ca7_0    conda-forge
pint                      0.22               pyhd8ed1ab_1    conda-forge
pip                       23.2.1             pyhd8ed1ab_0    conda-forge
platformdirs              3.10.0             pyhd8ed1ab_0    conda-forge
ply                       3.11                       py_1    conda-forge
pooch                     1.7.0              pyha770c72_3    conda-forge
prettytable               3.8.0              pyhd8ed1ab_0    conda-forge
prompt-toolkit            3.0.39             pyha770c72_0    conda-forge
prompt_toolkit            3.0.39               hd8ed1ab_0    conda-forge
psutil                    5.9.5           py310h8d17308_0    conda-forge
pthread-stubs             0.4               hcd874cb_1001    conda-forge
pthreads-win32            2.9.1                hfa6e2cd_3    conda-forge
pure_eval                 0.2.2              pyhd8ed1ab_0    conda-forge
pyface                    8.0.0              pyhd8ed1ab_0    conda-forge
pygments                  2.16.1             pyhd8ed1ab_0    conda-forge
pyparsing                 3.0.9              pyhd8ed1ab_0    conda-forge
pyqt                      5.15.9          py310h1fd54f2_4    conda-forge
pyqt5-sip                 12.12.2         py310h00ffb61_4    conda-forge
pysocks                   1.7.1              pyh0701188_6    conda-forge
python                    3.10.6          hcf16a7b_0_cpython    conda-forge
python-blosc              1.10.6          py310h394145c_1    conda-forge
python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
python_abi                3.10                    3_cp310    conda-forge
pyusid                    0.0.10             pyhd8ed1ab_1    conda-forge
pywavelets                1.4.1           py310h9b08ddd_0    conda-forge
pywin32                   304             py310h00ffb61_2    conda-forge
pyyaml                    6.0             py310h8d17308_5    conda-forge
pyzmq                     25.1.1          py310hcd737a0_0    conda-forge
qt-main                   5.15.8              h2c8576c_12    conda-forge
requests                  2.31.0             pyhd8ed1ab_0    conda-forge
scikit-image              0.20.0          py310h1c4a608_1    conda-forge
scikit-learn              1.3.0           py310hd266714_0    conda-forge
scipy                     1.11.1          py310h578b7cb_0    conda-forge
setuptools                68.0.0             pyhd8ed1ab_0    conda-forge
sidpy                     0.12.0             pyhd8ed1ab_0    conda-forge
sip                       6.7.11          py310h00ffb61_0    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
snappy                    1.1.10               hfb803bf_0    conda-forge
sortedcontainers          2.4.0              pyhd8ed1ab_0    conda-forge
sparse                    0.14.0             pyhd8ed1ab_0    conda-forge
stack_data                0.6.2              pyhd8ed1ab_0    conda-forge
sympy                     1.12               pyh04b8f61_3    conda-forge
tbb                       2021.10.0            h91493d7_0    conda-forge
tblib                     1.7.0              pyhd8ed1ab_0    conda-forge
threadpoolctl             3.2.0              pyha21a80b_0    conda-forge
tifffile                  2023.8.12          pyhd8ed1ab_1    conda-forge
tk                        8.6.12               h8ffe710_0    conda-forge
toml                      0.10.2             pyhd8ed1ab_0    conda-forge
tomli                     2.0.1              pyhd8ed1ab_0    conda-forge
toolz                     0.12.0             pyhd8ed1ab_0    conda-forge
tornado                   6.3.3           py310h8d17308_0    conda-forge
tqdm                      4.66.1             pyhd8ed1ab_0    conda-forge
traitlets                 5.9.0              pyhd8ed1ab_0    conda-forge
traits                    6.4.2           py310h8d17308_0    conda-forge
traitsui                  8.0.0              pyhd8ed1ab_0    conda-forge
typing-extensions         4.7.1                hd8ed1ab_0    conda-forge
typing_extensions         4.7.1              pyha770c72_0    conda-forge
tzdata                    2023c                h71feb2d_0    conda-forge
ucrt                      10.0.22621.0         h57928b3_0    conda-forge
unicodedata2              15.0.0          py310h8d17308_0    conda-forge
urllib3                   2.0.4              pyhd8ed1ab_0    conda-forge
vc                        14.3                h64f974e_17    conda-forge
vc14_runtime              14.36.32532         hfdfe4a8_17    conda-forge
vs2015_runtime            14.36.32532         h05e6639_17    conda-forge
wcwidth                   0.2.6              pyhd8ed1ab_0    conda-forge
werkzeug                  2.3.7              pyhd8ed1ab_0    conda-forge
wheel                     0.41.1             pyhd8ed1ab_0    conda-forge
widgetsnbextension        4.0.8              pyhd8ed1ab_0    conda-forge
win_inet_pton             1.1.0              pyhd8ed1ab_6    conda-forge
xorg-libxau               1.0.11               hcd874cb_0    conda-forge
xorg-libxdmcp             1.1.3                hcd874cb_0    conda-forge
xz                        5.2.6                h8d14728_0    conda-forge
yaml                      0.2.5                h8ffe710_2    conda-forge
zarr                      2.16.0             pyhd8ed1ab_0    conda-forge
zeromq                    4.3.4                h0e60522_1    conda-forge
zfp                       1.0.0                h63175ca_3    conda-forge
zict                      3.0.0              pyhd8ed1ab_0    conda-forge
zipp                      3.16.2             pyhd8ed1ab_0    conda-forge
zlib-ng                   2.0.7                hcfcfb64_0    conda-forge
zstd                      1.5.2                h12be248_7    conda-forge
```

#### Additional context
I'd be hugely grateful if anyone could suggest a fix!

**Comment by ericpre** [comment]
Can you try in the jupyter qtconsole, spyder or in spyder?
It will most likely be related to the execution of the notebook, for example, sometime running a cell of a notebook in jupyterlab hang after it has been executed and it need to be re-run... In the jupyter notebook application (as oppose to jupyterlab) it is better.
The libraries in jupyter stack evolves quickly and it is not straightforward to figure out what is happening because many libraries are involved!

**Comment by lnaglecocco** [comment]
Hi Eric,

Thanks for getting back to me.

I loaded up Idle from Anaconda Prompt while in my hyperspy environment, and I was able to use `hs.load()` from there. So I guess the issue is with Jupyter notebook. It's a bit odd as other things work fine in Jupyter. This works fine:

```
import matplotlib.pyplot as plt
plt.plot([1,2,3],[1,4,2])
```

And I tried doing things with Pymatgen and it works fine too. Just not Hyperspy.

I will play around with it and see if I can get it to work... if I find a solution I will comment it here

**Comment by lnaglecocco** [comment]
Hi Eric,
I haven't found a solution. I just tried it on a different computer (but one where I installed everything the same) and obtained the same result, regarding the cell that won't run.

Maybe worth noting that when I manually force the run to stop, I obtain the following error, which I don't know how to interpret. I don't understand how it's a "permission error"
```
---------------------------------------------------------------------------
PermissionError                           Traceback (most recent call last)
File C:\ProgramData\anaconda3\envs\hyperspy-env\lib\tempfile.py:256, in _mkstemp_inner(dir, pre, suf, flags, output_type)
    255 try:
--> 256     fd = _os.open(file, flags, 0o600)
    257 except FileExistsError:

PermissionError: [Errno 13] Permission denied: 'C:\\ProgramData\\anaconda3\\envs\\hyperspy-env\\lib\\site-packages\\hyperspy\\misc\\__pycache__\\tmpmhu8cum8'

During handling of the above exception, another exception occurred:

KeyboardInterrupt                         Traceback (most recent call last)
Cell In[4], line 1
----> 1 LowLoss_raw = hs.load("EELS Spectrum Image (low-loss).dm4")
      2 # HighLoss_raw = hs.load("EELS Spectrum Image (high-loss).dm4")
      3 # SurveyImage = hs.load("ADF Image (SI Survey).dm4")

File C:\ProgramData\anaconda3\envs\hyperspy-env\lib\site-packages\hyperspy\api.py:45, in __getattr__(name)
     43 if name in _import_mapping.keys():
     44     import_path = 'hyperspy' + _import_mapping.get(name)
---> 45     return getattr(importlib.import_module(import_path), name)
     46 else:
     47     return importlib.import_module("." + name, 'hyperspy')

File C:\ProgramData\anaconda3\envs\hyperspy-env\lib\importlib\__init__.py:126, in import_module(name, package)
    124             break
    125         level += 1
--> 126 return _bootstrap._gcd_import(name[level:], package, level)

File <frozen importlib._bootstrap>:1050, in _gcd_import(name, package, level)

File <frozen importlib._bootstrap>:1027, in _find_and_load(name, import_)

File <frozen importlib._bootstrap>:1006, in _find_and_load_unlocked(name, import_)

File <frozen importlib._bootstrap>:688, in _load_unlocked(spec)

File <frozen importlib._bootstrap_external>:883, in exec_module(self, module)

File <frozen importlib._bootstrap>:241, in _call_with_frames_removed(f, *args, **kwds)

File C:\ProgramData\anaconda3\envs\hyperspy-env\lib\site-packages\hyperspy\io.py:35
     33 from hyperspy.drawing.marker import markers_metadata_dict_to_markers
     34 from hyperspy.exceptions import VisibleDeprecationWarning
---> 35 from hyperspy.misc.io.tools import ensure_directory
     36 from hyperspy.misc.io.tools import overwrite as overwrite_method
     37 from hyperspy.misc.utils import strlist2enumeration

File C:\ProgramData\anaconda3\envs\hyperspy-env\lib\site-packages\hyperspy\misc\io\tools.py:25
     22 from pathlib import Path
     23 import os
---> 25 from hyperspy.misc.utils import DictionaryTreeBrowser
     28 _logger = logging.getLogger(__name__)
     31 def dump_dictionary(
     32     file, dic, string="root", node_separator=".", value_separator=" = "
     33 ):

File C:\ProgramData\anaconda3\envs\hyperspy-env\lib\site-packages\hyperspy\misc\utils.py:36
     33 import dask.array as da
     34 import numpy as np
---> 36 from hyperspy.misc.signal_tools import broadcast_signals
     37 from hyperspy.exceptions import VisibleDeprecationWarning, LazyCupyConversion
     38 from hyperspy.docstrings.signal import SHOW_PROGRESSBAR_ARG

File C:\ProgramData\anaconda3\envs\hyperspy-env\lib\site-packages\hyperspy\misc\signal_tools.py:25
     22 import numpy as np
     24 from hyperspy.misc.axis_tools import check_axes_calibration
---> 25 from hyperspy.misc.array_tools import are_aligned
     27 _logger = logging.getLogger(__name__)
     30 def _get_shapes(am, ignore_axis):

File C:\ProgramData\anaconda3\envs\hyperspy-env\lib\site-packages\hyperspy\misc\array_tools.py:195
    190 # Replacing space is necessary to get the correct indentation
    191 rebin.__doc__ %= REBIN_ARGS.replace("        ", "    ")
    194 @njit(cache=True)
--> 195 def _linear_bin_loop(result, data, scale):  # pragma: no cover
    196     for j in range(result.shape[0]):
    197         # Begin by determining the upper and lower limits of a given new pixel.
    198         x1 = j * scale

File C:\ProgramData\anaconda3\envs\hyperspy-env\lib\site-packages\numba\core\decorators.py:234, in _jit.<locals>.wrapper(func)
    230 disp = dispatcher(py_func=func, locals=locals,
    231                   targetoptions=targetoptions,
    232                   **dispatcher_args)
    233 if cache:
--> 234     disp.enable_caching()
    235 if sigs is not None:
    236     # Register the Dispatcher to the type inference mechanism,
    237     # even though the decorator hasn't returned yet.
    238     from numba.core import typeinfer

File C:\ProgramData\anaconda3\envs\hyperspy-env\lib\site-packages\numba\core\dispatcher.py:863, in Dispatcher.enable_caching(self)
    862 def enable_caching(self):
--> 863     self._cache = FunctionCache(self.py_func)

File C:\ProgramData\anaconda3\envs\hyperspy-env\lib\site-packages\numba\core\caching.py:601, in Cache.__init__(self, py_func)
    599 self._name = repr(py_func)
    600 self._py_func = py_func
--> 601 self._impl = self._impl_class(py_func)
    602 self._cache_path = self._impl.locator.get_cache_path()
    603 # This may be a bit strict but avoids us maintaining a magic number

File C:\ProgramData\anaconda3\envs\hyperspy-env\lib\site-packages\numba\core\caching.py:333, in CacheImpl.__init__(self, py_func)
    331 source_path = inspect.getfile(py_func)
    332 for cls in self._locator_classes:
--> 333     locator = cls.from_function(py_func, source_path)
    334     if locator is not None:
    335         break

File C:\ProgramData\anaconda3\envs\hyperspy-env\lib\site-packages\numba\core\caching.py:180, in _SourceFileBackedLocatorMixin.from_function(cls, py_func, py_file)
    178 self = cls(py_func, py_file)
    179 try:
--> 180     self.ensure_cache_path()
    181 except OSError:
    182     # Cannot ensure the cache directory exists or is writable
    183     return

File C:\ProgramData\anaconda3\envs\hyperspy-env\lib\site-packages\numba\core\caching.py:107, in _CacheLocator.ensure_cache_path(self)
    105 os.makedirs(path, exist_ok=True)
    106 # Ensure the directory is writable by trying to write a temporary file
--> 107 tempfile.TemporaryFile(dir=path).close()

File C:\ProgramData\anaconda3\envs\hyperspy-env\lib\tempfile.py:559, in NamedTemporaryFile(mode, buffering, encoding, newline, suffix, prefix, dir, delete, errors)
    557     return fd
    558 try:
--> 559     file = _io.open(dir, mode, buffering=buffering,
    560                     newline=newline, encoding=encoding, errors=errors,
    561                     opener=opener)
    562     try:
    563         raw = getattr(file, 'buffer', file)

File C:\ProgramData\anaconda3\envs\hyperspy-env\lib\tempfile.py:556, in NamedTemporaryFile.<locals>.opener(*args)
    554 def opener(*args):
    555     nonlocal name
--> 556     fd, name = _mkstemp_inner(dir, prefix, suffix, flags, output_type)
    557     return fd

File C:\ProgramData\anaconda3\envs\hyperspy-env\lib\tempfile.py:262, in _mkstemp_inner(dir, pre, suf, flags, output_type)
    258     continue    # try again
    259 except PermissionError:
    260     # This exception is thrown when a directory with the chosen name
    261     # already exists on windows.
--> 262     if (_os.name == 'nt' and _os.path.isdir(dir) and
    263         _os.access(dir, _os.W_OK)):
    264         continue
    265     else:

File C:\ProgramData\anaconda3\envs\hyperspy-env\lib\genericpath.py:42, in isdir(s)
     40 """Return true if the pathname refers to an existing directory."""
     41 try:
---> 42     st = os.stat(s)
     43 except (OSError, ValueError):
     44     return False

KeyboardInterrupt: 
```

**Comment by lnaglecocco** [comment]
I don't know if it's at all helpful, but I get an identical error when using JupyterLab. Though like I said, it works ok in Idle run from Anaconda Prompt

**Comment by CSSFrancis** [comment]
Hi @lnaglecocco I'm not the most familiar with windows systems but it could be a permissions error.  I think that because it is trying to access a file in the ProgramData Folder you might need admin access?  

If you have admin permissions you should be able to launch anaconda with admin access and that should help things out.  Otherwise you might have to install anaconda in a different place on your computer like in your home directory.

**Comment by EoinScope** [comment]
Hi, I had an issue with hs.load() today after not using Hyperspy for a while and found this thread. Rather than the cell getting stuck in an infinite loop though I got an error message when hs.load() was trying to determine that the module snappy was greater than v0.53 by checking the command snappy.compress(). I had the same version 1.1.10 of snappy as installed in the module list above but there is no command snappy.compress() in that module.

I fixed the problem by installing python-snappy instead: pip install python-snappy

After this hs.load() worked correctly for me. Maybe you are experiencing a different issue since you didn't get the same error but hopefully this helps.

**Comment by lnaglecocco** [comment]
Hi @CSSFrancis - your suggestion worked. I open Jupyter by right-clicking and selecting "Run as administrator" and no longer encounter issues.

@EoinScope thanks for the suggestion

**Comment by CSSFrancis** [comment]
Glad this worked out!

**Comment by ericpre** [comment]
Regarding the permission issue, anaconda [recommend to avoid installing for all users](https://docs.anaconda.com/free/anaconda/install/windows/), I don't know why but in the past, I did notice that all users installation does work well, so I avoid it when not necessary! :)
From the error it is surprising that numba cache is causing issue because it has [fallbacks](https://github.com/numba/numba/blob/bd7ebcfd4b850208b627a3f75d4706000be36275/numba/core/caching.py#L318-L321) supposed to deal with it and in this case it should fall back to your user's `AppData/Local` folder.

**Comment by HanHsuanWu** [comment]
Just following up so was there a better solution then opening anaconda with "Run as administrator" every time? Inconveniently, start_jupyter_cm only opens jupyter notebook normally which result in the numba cache permissions error.

**Comment by HanHsuanWu** [comment]
Yea so the best solution for this problem on windows is to just install anaconda (or miniconda) for "Just Me." In this case, files are installed in \Users (instead of \ProgramData) and will not require admin privileges to write, which happens when creating cache files. It is painful but just re-installing miniconda solved the problem for me.

**Comment by ericpre** [comment]
Indeed single user install are better/more simple and this is valid for any python distribution and Anaconda, conda-forge do recommend it. The hyperspy-bundle does [recommend it as well](https://hyperspy.org/hyperspy-bundle/install.html#windows), but maybe it would be worth making a note or warning in the user guide to put more emphasis on it?

That being said in this specific case, the issue with numba cache shouldn't occur as [mentioned above](https://github.com/hyperspy/hyperspy/issues/3212#issuecomment-1690564805) and there may be a bug in numba too.

**Comment by HanHsuanWu** [comment]
The same issue reported at numba 8 years ago is here:
https://github.com/numba/numba/issues/1771

**Comment by ericpre** [comment]
> The same issue reported at numba 8 years ago is here: [numba/numba#1771](https://github.com/numba/numba/issues/1771)

Yes, but it should have been fixed in https://github.com/numba/numba/pull/1772 which was included in numba 0.25.0, maybe a regression from numba side, that they are not aware of.
